sdl_android/src/androidTest/java/com/smartdevicelink/protocol/SdlProtocolTests.java

```diff
@@ -0,0 +1,465 @@
+package com.smartdevicelink.protocol;
+
+import android.test.AndroidTestCase;
+import android.util.Log;
+
+import com.smartdevicelink.SdlConnection.SdlConnection;
+import com.smartdevicelink.protocol.enums.SessionType;
+import com.smartdevicelink.protocol.SdlProtocol.MessageFrameAssembler;
+import com.smartdevicelink.security.SdlSecurityBase;
+import com.smartdevicelink.streaming.video.VideoStreamingParameters;
+import com.smartdevicelink.test.SampleRpc;
+import com.smartdevicelink.test.SdlUnitTestContants;
+import com.smartdevicelink.transport.BaseTransportConfig;
+import com.smartdevicelink.transport.MultiplexTransportConfig;
+import com.smartdevicelink.transport.RouterServiceValidator;
+
+import junit.framework.Assert;
+
+import java.io.ByteArrayOutputStream;
+import java.lang.reflect.Method;
+import java.util.List;
+
+public class SdlProtocolTests  extends AndroidTestCase {
+
+    int max_int = 2147483647;
+    byte[] payload;
+    MultiplexTransportConfig config;
+    SdlProtocol protocol;
+
+    ISdlProtocol defaultListener = new ISdlProtocol() {
```

Comment:
This can be mocked like this: https://github.com/smartdevicelink/sdl_android/blob/fe9a8db5b9e38bde9bd39f5f094d825b68ea3de1/sdl_android/src/androidTest/java/com/smartdevicelink/api/VideoStreamingManagerTests.java#L97

Reply:
The required dependencies haven't been added to develop yet. For the sake of merging I'd recommend waiting until all the large feature branches are merged and will go back and fix this.